hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -357,6 +357,12 @@ private Constants() {
    * This makes it different from {@link #REQUEST_TIMEOUT}, which is for total
    * HTTP request.
    * <p>
+   * In V1 SDK, this value was not used in POST/PUT operations; a low value
+   * (default 15s) allowed for a fast failure of all other operations.
+   * In V2 SDK this now applies to all operations, including uploads.
+   * A large timeout is now needed, even though it means that some service/network
+   * failures may now take a long time to surface.
```

Comment:
However, 15minutes sounds like too big as a timeout, doesn't it? What is the background of this huge value?

Technically, `15m` means `1 hour` when we retry 3 times.

Reply:
problem is we want a duration which is valid with many other streams competing for that bandwidth, so a single block upload of, say, 64M needs time. I don't know what a good time is.

Reply:
we could maybe cut to something a bit lower, like say 10m?

side issue: why don't you try it as is for a remote upload of the rc1 tar.gz and see how you get on without this patch.